docs/src/proposals/bidirectional_quic_notification_service.md

```diff
@@ -0,0 +1,166 @@
+to use notify in all the essential parts of the code.
+
+Example:
+```rust
+if transaction.is_err() {
+    notify!(transaction.err().to_string(),
+        NotificationType::NotifyTransaction{signature},
+        NotificationType::BankingStage,
+    );
+}
+```
+
+We can then create a new implementation of a connection pool where instead of just connecting to
+the next N leaders, it will also connect to the bidirectional services of these leaders.
+Before sending the transaction, it will subscribe to the `NotifyTransaction` notification
+and then send the transaction to the leaders. Each leader will then reply about how they handled
+the transactions, and we can easily determine the transaction's lifecycle.
+We added a mechanism on the client side called a bidirectional reply handler which will take care
+of the notifications received by each node client has subscribed to, and forward the notifications
+to the output channel.
+
+To avoid loops and over-notifications, validators will not create bidirectional channels with other
+validators to get notifications. The size of the reply message will be a fixed length of N bytes
+so that the client will know exactly that it has received a notification from the server after
+receiving N bytes. The messages will be serialized in binary before sending over the network.
+Here is the proposed structure:
+
+```rust
+pub const QUIC_REPLY_MESSAGE_SIZE: usize = 256;
+#[derive(Clone, Serialize, Deserialize)]
+pub enum QuicNotification {
+    TransactionMessage {
+        sender_identity: Pubkey,
+        transaction_signature: Signature,
+        message: Vec<u8>, // Is resized to 128 chars in constructor
+        approximate_slot: Slot,
+        padding: [u8; ] // rest
+    },
+}
+```
+
+In the above enum, is the definition of `QuicNotification` for a transaction.
+This enum will be serialized using `bincode` and sent over the network to the client.
+The message length is exactly 128 chars. A serialized message for type `QuicNotification`
+should have the exact length of `QUIC_NOTIFICATION_MESSAGE_SIZE`, which is set to 256 bytes at this time.
+If the message is larger than 128 characters, it will be truncated; if it is smaller, the rest of
+the characters will be set to `0`.
+
+We can additionally add limits so that this feature could not be used as an attack vector.
+
+1. Number of concurrent connections in bidirectional stream is limited to 5.
+2. Additionally server will close the connection when client closes the connection.
+3. Max number of subscriptions per connection : 10000
+4. Number of messages in the queue to be sent limited to : 10000
+
+We should also add a way to advertise the port for bidirectional message service. It could be fixed
+to `QUIC TPU PORT + SOME PREDEFINED OFFSET`.
+
+## In working
+
+We have already implemented this proposal for test purpose in v1.14.
+<https://github.com/solana-labs/solana/pull/29954>
```

Comment:
Can remove this, a proposal really should stand alone and not reference an implementation :)